src/CMakeLists.txt

```diff
@@ -130,10 +103,6 @@ set (NN_SOURCES
     utils/sleep.c
     utils/thread.h
     utils/thread.c
-    utils/thread_posix.h
```

Comment:
In this case, maybe rearchitecting the stuff *now* is something to wait
for; keep the existing solution used in autoconf and old cmake (which seems
to work well enough), and we can explore alternative better ideas in the
future.  This would have the benefit of reducing the scope of change
required for cmake support.

On Mon, Apr 18, 2016 at 1:31 PM, Jack R. Dunaway <notifications@github.com>
wrote:

> In src/CMakeLists.txt
> <https://github.com/nanomsg/nanomsg/pull/636#discussion_r60128764>:
>
> > @@ -130,10 +103,6 @@ set (NN_SOURCES
> >      utils/sleep.c
> >      utils/thread.h
> >      utils/thread.c
> > -    utils/thread_posix.h
>
> The reason I chose this method is to prune other-platform source code from
> the backend-generated project file -- in my case this is a Visual Studio
> Solution. By omitting, e.g., worker_posix.inc, this means when I search
> for symbols, it doesn't return duplicates in non-relevant platforms -- only
> worker_win.inc was included and relevant. It makes the IDE experience
> better.
>
> That said ... there's merit in understanding other platforms, even if it's
> not relevant to your own current platform. In which case, I would prefer a
> rearchitecture of the source code to be along the lines of this:
> https://github.com/nanomsg/nanomsg/blob/master/src/utils/clock.c#L71-L121
>
> Either way ... much of this entire CMake rearchitecture is a stepping
> stone to get from where we are to where we want to be, and in the meantime,
> it's likely to carry along some cruft that can then be hit later.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/nanomsg/nanomsg/pull/636/files/95b664f66e89dfbcc99877d3132a54b420c69cf5#r60128764>
>